src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,109 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+import java.io.FileWriter;
+import java.io.IOException;
+
+class Duke {
+
+    Scanner myScanner = new Scanner(System.in);
+
+
+    private Parser parser;
+
+
+    public Duke(String filePath) {
+        parser = new Parser(myScanner);
+    }
+
+    static List<Task> toDo = new ArrayList<>();
+
+    static int countTasks() {
+        int count = 0;
+        for (Task task : toDo) {
+            if (!task.isDone) {
+                count++;
+            }
+        }
+        return count;
+    }
+
+    static void saveTasks(List<Task> tasks) {//Called on "bye"
+        try {
+            File data_file = new File("Data/DukeData.txt");
+            FileWriter writer = new FileWriter("Data/DukeData.txt");//Overwriting entire file
+            for (Task task : tasks) {
+                writer.write(task.toString());
+                writer.write("\n");
+            }
+            writer.close();
+            if (data_file.createNewFile()) {
+                System.out.println("File created: " + data_file.getName());
+            } else {
+                System.out.println("File already exists.");
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    static void readFile(String filePath) throws FileNotFoundException {
+        File f = new File(filePath); // create a File for the given file path
+        Scanner s = new Scanner(f); // create a Scanner using the File as the source
+        while (s.hasNext()) {
+            String currLine = s.nextLine();
+//            System.out.println(currLine);
+            char taskType = currLine.charAt(1);
+//            System.out.println(taskType);
+            switch (taskType) {
+                case 'T':
```

Comment:
i believe case statements should not be indented according to our modules coding standards :)